### PR TITLE
Integrate new FFT iterative algorithms and tidy-up the FFT API

### PR DESCRIPTION
**Pull request by xqft**
# Integrate new FFT algorithms and tidy-up the FFT API

## Description

This is a follow up of #127. The changes remove the previous recursive FFT algorithm and integrates the new, faster ones. Also adds new methods for twiddle/roots of unity generation and tidies-up the API in general.

This is part of our efforts for #116.

## Type of change

Please delete options that are not relevant.

- [X] New feature
- [X] Optimization

## Checklist
- [X] Unit tests added
- [X] This change requires new documentation.
  - [X] Documentation has been added/updated.

### CODE REVIEW

**Review by jrchatruc — APPROVED** [review]

**Review by IAvecilla — APPROVED** [review]
Just some minor details but everything looks good

File: math/src/field/traits.rs [hunk 1]
```diff
@@ -16,25 +28,63 @@ pub trait IsTwoAdicField: IsField {
     const TWO_ADIC_PRIMITVE_ROOT_OF_UNITY: Self::BaseType;
     const GENERATOR: Self::BaseType;
 
-    /// Returns the primitive root of unity of order 2^k.
-    fn get_root_of_unity(k: u64) -> Result<FieldElement<Self>, FFTError> {
+    /// Returns the primitive root of unity of order $2^{order}$.
+    fn get_root_of_unity(order: u64) -> Result<FieldElement<Self>, FFTError> {
```
[the remark this answers is not in the document]
Comment (xqft) [reply math/src/field/traits.rs]: Done already, based on our previous conversations

File: math/src/field/traits.rs [hunk 2]
```diff
@@ -16,25 +28,63 @@ pub trait IsTwoAdicField: IsField {
     const TWO_ADIC_PRIMITVE_ROOT_OF_UNITY: Self::BaseType;
     const GENERATOR: Self::BaseType;
```
[the remark this answers is not in the document]
Comment (xqft) [reply math/src/field/traits.rs:29]: I'm not sure if this wouldn't be required for some math thing related to two-adic fields we are missing, @jrchatruc wdyt?

[the remark this answers is not in the document]
Comment (xqft) [reply math/src/field/traits.rs:29]: Resolved in private chats

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**